File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> alphabet(26,0);

        if(s.size()!=t.size()) return false;

        for(int i=0;i<s.size();i++){
            alphabet[s[i]-'a']++;
            alphabet[t[i]-'a']--;
        }

        for(int i=0;i<alphabet.size();i++){
            if(alphabet[i]!=0 ) return false;
        }

        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        elements = {}
        for i in s:
            elements[i]=elements.get(i,0)+1
        
        for i in t:
            elements[i]=elements.get(i,0)-1
        
        for j in elements.values():
            if j!=0:
                return False
        
        return True
        
        